### PR TITLE
Integration test to check health check fix

### PR DESCRIPTION
**Pull request by Rohit-PX**
Signed-off-by: Rohit-PX <rohit@portworx.com>

**What type of PR is this?**
integration-test

**What this PR does / why we need it**:
Test this fix:  https://github.com/libopenstorage/stork/releases/tag/v2.3.4

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
no

**Does this change need to be cherry-picked to a release branch?**:
2.4

### CODE REVIEW

File: test/integration_test/health_monitor_test.go [hunk 1]
```diff
@@ -115,3 +117,74 @@ func stopKubeletTest(t *testing.T) {
 	destroyAndWait(t, ctxs)
 
 }
+
+func healthCheckFixTest(t *testing.T) {
+	// Cordon node where the test is running. This is so that we don't end up stopping
```
[the remark this answers is not in the document]
Comment (Rohit-PX) [reply test/integration_test/health_monitor_test.go:122]: Removed.

File: test/integration_test/health_monitor_test.go [hunk 2]
```diff
@@ -115,3 +117,74 @@ func stopKubeletTest(t *testing.T) {
...
+
+	defer func() {
+		if len(testPodNode) > 0 {
+			err = core.Instance().UnCordonNode(testPodNode, defaultWaitTimeout, defaultWaitInterval)
+			require.NoError(t, err, "Error uncordorning k8s node for stork test pod")
+		}
+	}()
+	logrus.Infof("Step: Completed cordoning node: %s", testPodNode)
+
+	// When a node's storage is offline stork should not bouncing pods right away.
```
[the remark this answers is not in the document]
Comment (Rohit-PX) [reply test/integration_test/health_monitor_test.go:140]: Fixed.

File: test/integration_test/health_monitor_test.go [hunk 3]
```diff
@@ -115,3 +117,74 @@ func stopKubeletTest(t *testing.T) {
...
+	logrus.Infof("Step: Started volume driver again on node: %s", scheduledNodes[0].Name)
+
+	// Verify that app comes up on the same node
+	scheduledNodesPostStop, err := schedulerDriver.GetNodesForApp(ctxs[0])
+	require.NoError(t, err, "Error getting node for app")
+	logrus.Infof("Step: App scheduled on node after restart: %s", scheduledNodes[0].Name)
+
+	require.Equal(t, 1, len(scheduledNodesPostStop), "App should be scheduled on one node")
+	require.Equal(t, stoppedNode.Name, scheduledNodesPostStop[0].Name,
+		"App scheduled on a different node after volume driver stopped for less than a minute")
```
[the remark this answers is not in the document]
Comment (Rohit-PX) [reply test/integration_test/health_monitor_test.go:188]: Added check for Pod UIDs before and after volume driver restart.

File: test/integration_test/health_monitor_test.go [hunk 4]
```diff
@@ -115,3 +119,87 @@ func stopKubeletTest(t *testing.T) {
...
+	// Get uuid for the app
+	var preUIDList, postUIDList []string
+	for _, spec := range ctxs[0].App.SpecList {
+		if dep, ok := spec.(*appsapi.Deployment); ok {
+			depPods, err := apps.Instance().GetDeploymentPods(dep)
+			require.NoError(t, err, "Error getting pods for deployment ,mysql.")
+			for _, pod := range depPods {
+				uid := pod.UID
+				preUIDList = append(preUIDList, string(uid))
+				logrus.Infof("RK=> UID for pod %s: %v", pod.Name, preUIDList)
```
Comment (disrani-px) [inline test/integration_test/health_monitor_test.go:144]: Remove debug log

File: test/integration_test/health_monitor_test.go [hunk 5]
```diff
@@ -115,3 +119,87 @@ func stopKubeletTest(t *testing.T) {
...
+
+	// verify the app has not restarted after volume driver was stopped, by comparing the start time
+	for _, spec := range ctxs[0].App.SpecList {
+		if dep, ok := spec.(*appsapi.Deployment); ok {
+			depPods, err := apps.Instance().GetDeploymentPods(dep)
+			require.NoError(t, err, "Error getting pods for deployment ,mysql.")
+			for _, pod := range depPods {
+				uid := pod.UID
+				postUIDList = append(postUIDList, string(uid))
+				logrus.Infof("RK=> UID for pod %s: %v", pod.Name, postUIDList)
```
Comment (disrani-px) [inline test/integration_test/health_monitor_test.go:193]: Remove debug logs

File: test/integration_test/health_monitor_test.go [hunk 6]
```diff
@@ -115,3 +119,87 @@ func stopKubeletTest(t *testing.T) {
...
+	ctxs, err := schedulerDriver.Schedule(generateInstanceID(t, "stopdrivertest"),
+		scheduler.ScheduleOptions{AppKeys: []string{"mysql-1-pvc"}})
+	require.NoError(t, err, "Error scheduling task")
+	require.Equal(t, 1, len(ctxs), "Only one task should have started")
+
+	err = schedulerDriver.WaitForRunning(ctxs[0], defaultWaitTimeout, defaultWaitInterval)
+	require.NoError(t, err, "Error waiting for pod to get to running state")
+
+	// Get uuid for the app
+	var preUIDList, postUIDList []string
```
Comment (disrani-px) [inline test/integration_test/health_monitor_test.go:136]: Better to keep a map of podName->podUID. Adding to a map is more efficient than appending to a slice in golang

**Comment by Rohit-PX** [comment]
Ran here successfully.
https://jenkins.portworx.co/job/Stork/view/Stork%20master%20Dashboard/job/stork-2-dev-test/73/console

**Pull request merged**